pkgs/applications/editors/vscode/generic.nix

```diff
@@ -108,6 +111,30 @@ let
       )
     '';
 
+    # See https://github.com/NixOS/nixpkgs/issues/49643#issuecomment-873853897
+    postPatch = ''
+      # PATCHING RUN AS SUDO
+      # --------------------
+      # where to find the node.js librar .asar was described here https://github.com/NixOS/nixpkgs/issues/76526#issuecomment-569131432
+      packed="resources/app/node_modules.asar"
+      # we unpack directly into the same name without .asar ending,
+      # which is adapted from hacking-skype-tutorial https://www.codepicky.com/hacking-electron-restyle-skype/
+      unpacked="resources/app/node_modules"
+
+      ${nodePackages.asar}/bin/asar extract "$packed" "$unpacked"
+      # we change paths to pkexec and bash
+      # as described here https://github.com/NixOS/nixpkgs/blob/fc553c0bc5411478e2448a707f74369ae9351e96/pkgs/tools/misc/etcher/default.nix#L49y
+      sed -i "
```

Comment:
Couldn't you use `patchShebangs`?

Reply:
If I only know what is it and how to do that.

In fact I've just PRed a very useful patch I found around as is.

Reply:
> In fact I've just PRed a very useful patch I found around as is.

It's great that you created a PR so the issue will be fixed for everyone (and not just those that adapt the patch into their overlay). If it's added to nixpkgs though, the code quality has to be high, so I'm trying to help you achieve a high quality of the PR.

> If I only know what is it and how to do that.

`patchShebangs` replaces paths like `/usr/bin/something` with the appropriate nix store path like `${pkgs.something}/bin/something`. It's used in a lot of places in nixpkgs [as one can see in the search](https://github.com/NixOS/nixpkgs/search?q=patchShebangs). See there for examples.

I think that it might be able to fix this in a more standard way.

Reply:
patchshebangs only replaces the shebang !#

Reply:
Yeah, in this case you want substituteInPlace